spring-boot-project/spring-boot-test/src/main/java/org/springframework/boot/test/context/SpringBootConfigurationFinder.java

```diff
@@ -29,21 +30,34 @@
 import org.springframework.util.ClassUtils;
 
 /**
- * Internal utility class to scan for a {@link SpringBootConfiguration} class.
+ *  Utility class to scan for a {@link SpringBootConfiguration} or custom annotation.
  *
  * @author Phillip Webb
+ * @author Artsiom Yudovin
+ * @since 2.1.0
  */
-final class SpringBootConfigurationFinder {
+public final class SpringBootConfigurationFinder {
 
 	private static final Map<String, Class<?>> cache = Collections
 			.synchronizedMap(new Cache(40));
 
 	private final ClassPathScanningCandidateComponentProvider scanner;
 
-	SpringBootConfigurationFinder() {
+	/**
+	 * Default constructor initializing finder to scan for a {@link SpringBootConfiguration} annotation.
+	 */
+	public SpringBootConfigurationFinder() {
+		this(SpringBootConfiguration.class);
+	}
+
+	/**
+	 * Customiable constructor allow to provide the custom annotation to scan for.
```

Comment:
typo? "Customizable" or "Custom" ...

Reply:
Thanks @kopa I am polishing this contribution and I've already rephrased this.

Reply:
@snicoll Thank you for your help